bridges/snowbridge/primitives/core/src/outbound/v2.rs

```diff
@@ -0,0 +1,348 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+//! # Outbound V2 primitives
+
+use crate::outbound::{OperatingMode, SendError, SendMessageFeeProvider};
+use alloy_sol_types::sol;
+use codec::{Decode, Encode};
+use frame_support::{pallet_prelude::ConstU32, BoundedVec};
+use hex_literal::hex;
+use scale_info::TypeInfo;
+use sp_arithmetic::traits::{BaseArithmetic, Unsigned};
+use sp_core::{RuntimeDebug, H160, H256};
+use sp_std::{vec, vec::Vec};
+
+use alloy_primitives::{Address, FixedBytes};
+use alloy_sol_types::SolValue;
+
+sol! {
+	struct InboundMessageWrapper {
```

Comment:
Is it possible to use modules as namespaces instead of appending `Wrapper` to the struct names? Thinking of this pattern:

```rust
mod abi {
	sol! {
		struct InboundMessageWrapper { ... }
	}
}
```

Then we can reference the type using `abi::InboundMessage`

Reply:
https://github.com/yrong/polkadot-sdk/pull/4/commits/83b6ff851625be9c5a520bdf602607c28e4de668

Reply:
The point of adding the `abi` namespace is that we rename `InboundMessageWrapper` and `CommandWrapper` to `InboundMessage` and `Command` respectively

Reply:
Yeah, it seems a bit cubersome. The reason is that I can't add macro `#[derive(Encode, Decode, RuntimeDebug, PartialEq,TypeInfo)]` before `InboundMessageWrapper` which is required for storing `InboundMessage` on chain [here](https://github.com/yrong/polkadot-sdk/blob/cff1cba3dfc21d2be621eae984b83e4e0963c81b/bridges/snowbridge/pallets/outbound-queue-v2/src/lib.rs#L360-L361), error as follows:


```
--> /Users/yangrong/Projects/polkadot-sdk/bridges/snowbridge/primitives/core/src/outbound/v2.rs:31:12
     |
  31 |         #[derive(Encode, Decode, RuntimeDebug, PartialEq,TypeInfo)]
     |                  ^^^^^^ the trait `WrapperTypeEncode` is not implemented for `alloy_primitives::FixedBytes<32>`, which is required by `alloy_primitives::FixedBytes<32>: Encode`
     |
     = help: the following other types implement trait `WrapperTypeEncode`:
               &T
               &mut T
               Arc<T>
               Box<T>
               Cow<'a, T>
               Rc<T>
               alloy_primitives::bytes::Bytes
               parity_scale_codec::Ref<'a, T, U>
             and 3 others
     = note: required for `alloy_primitives::FixedBytes<32>` to implement `Encode`
     = note: this error originates in the derive macro `Encode` (in Nightly builds, run with -Z macro-backtrace for more info)

  error[E0277]: the trait bound `alloy_primitives::FixedBytes<32>: Decode` is not satisfied
    --> /Users/yangrong/Projects/polkadot-sdk/bridges/snowbridge/primitives/core/src/outbound/v2.rs:34:4
     |
  34 |             bytes32 origin;
     |             ^^^^^^^ the trait `WrapperTypeDecode` is not implemented for `alloy_primitives::FixedBytes<32>`, which is required by `alloy_primitives::FixedBytes<32>: Decode`
     |
     = help: the following other types implement trait `WrapperTypeDecode`:
               Arc<T>
               Box<T>
               Rc<T>
               sp_core::Bytes
     = note: required for `alloy_primitives::FixedBytes<32>` to implement `Decode`

  error[E0277]: the trait bound `alloy_primitives::FixedBytes<32>: TypeInfo` is not satisfied
     --> /Users/yangrong/Projects/polkadot-sdk/bridges/snowbridge/primitives/core/src/outbound/v2.rs:34:4
      |
  31  |         #[derive(Encode, Decode, RuntimeDebug, PartialEq,TypeInfo)]
      |                                                          -------- required by a bound introduced by this call
  ...
  34  |             bytes32 origin;
      |             ^^^^^^^ the trait `TypeInfo` is not implemented for `alloy_primitives::FixedBytes<32>`

```


It's a workaround why I add some intermediate data structure here.